FABulous/FABulous.py

```diff
@@ -67,24 +67,86 @@ def setup_logger(verbosity: int):
     logger.add(sys.stdout, format=log_format, level="DEBUG", colorize=True)
 
 
-metaDataDir = ".FABulous"
+def setup_global_env_vars(args: argparse.Namespace) -> None:
+    """
+    Set up global  environment variables
 
-fabulousRoot = os.getenv("FAB_ROOT")
-if fabulousRoot is None:
-    fabulousRoot = os.path.dirname(os.path.realpath(__file__))
-    logger.warning("FAB_ROOT environment variable not set!")
-    logger.warning(f"Using {fabulousRoot} as FAB_ROOT")
-else:
-    if not os.path.exists(fabulousRoot):
-        logger.error(
-            f"FAB_ROOT environment variable set to {fabulousRoot} but the directory does not exist"
-        )
-        sys.exit()
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Command line arguments
+    """
+    # Set FAB_ROOT environment variable
+    fabulousRoot = os.getenv("FAB_ROOT")
+    if fabulousRoot is None:
+        os.environ["FAB_ROOT"] = str(Path(__file__).resolve().parent)
+        logger.info("FAB_ROOT environment variable not set!")
+        logger.info(f"Using {fabulousRoot} as FAB_ROOT")
     else:
-        if os.path.exists(f"{fabulousRoot}/FABulous"):
-            fabulousRoot = f"{fabulousRoot}/FABulous"
+        # If there is the FABulous folder in the FAB_ROOT, then set the FAB_ROOT to the FABulous folder
+        if Path(fabulousRoot).exists():
+            if Path(fabulousRoot).joinpath("FABulous").exists():
+                fabulousRoot = str(Path(fabulousRoot).joinpath("FABulous"))
+            os.environ["FAB_ROOT"] = fabulousRoot
+        else:
+            logger.error(
+                f"FAB_ROOT environment variable set to {fabulousRoot} but the directory does not exist"
+            )
+            sys.exit()
+
+        logger.info(f"FAB_ROOT set to {fabulousRoot}")
+
+    # Load the .env file and make env variables available globally
+    if args.globalDotEnv:
+        gde = Path(args.globalDotEnv)
+        if gde.is_file():
+            load_dotenv(gde)
+            logger.info(f"Loaded global .env file from {gde}")
+        elif gde.joinpath(".env").exists() and gde.joinpath(".env").is_file():
+            load_dotenv(gde.joinpath(".env"))
+            logger.info(f"Loaded global .env file from {gde.joinpath('.env')}")
+        else:
+            logger.info(f"No global .env file found at {gde}")
+    elif (
+        Path(os.getenv("FAB_ROOT")).joinpath(".env").exists()
+        and Path(os.getenv("FAB_ROOT")).joinpath(".env").is_file()
+    ):
+        load_dotenv(Path(os.getenv("FAB_ROOT")).joinpath(".env"))
+        logger.info(f"Loaded global .env file from {fabulousRoot}/.env")
+    else:
+        logger.info("No global .env file found")
```

Comment:
Should terminate with error logging since reaching this point means something about the installation is wrong

Reply:
I would maybe make this a warning, since you could also use FABulous it entirely without a .env file, since we have Default values for everything, and you can set everything through environment variables in your terminal.

Reply:
That will do as well